src/h5web/visualizations/containers/NxSpectrumContainer.tsx

```diff
@@ -8,20 +8,26 @@ import { DimensionMapping } from '../../dimension-mapper/models';
 import MappedLineVis from '../line/MappedLineVis';
 import { ProviderContext } from '../../providers/context';
 import {
-  getSignalDataset,
+  getAttributeValue,
   getLinkedEntity,
   getAxesLabels,
 } from '../nexus/utils';
 import { VisContainerProps } from './models';
+import { assertStr } from '../shared/utils';
 
 function NxSpectrumContainer(props: VisContainerProps): ReactElement {
   const { entity } = props;
   assertGroup(entity);
 
   const { metadata } = useContext(ProviderContext);
-  const signalDataset = getSignalDataset(entity, metadata);
+  const signal = getAttributeValue(entity, 'signal');
+  if (!signal) {
+    throw new Error('Entity has no signal attribute');
+  }
+  assertStr(signal);
```

Comment:
`assertStr` is enough to check to `undefined`, so I don't think we need the throw. If the string is empty, `getLinkedEntity` will return `undefined` anyway.

Reply:
This makes me realise that we shouldn't throw inside `getSignalDataset` since it's now only called from the `supportsEntity` functions, and these functions are not equipped to catch errors. 

```ts
  if (!signal) {
    return undefined;
  }

  assertStr(signal);
```

... should be replaced with:

```ts
  if (typeof signal !== 'string') {
    return undefined;
  }
```

Perhaps we should get rid of `getSignalDataset` altogether and inline the code?